src/fshtypes/rules/AssignmentRule.ts

```diff
@@ -27,4 +31,29 @@ export class AssignmentRule extends Rule {
   get constructorName() {
     return 'AssignmentRule';
   }
+
+  toFSH(): string {
+    let printableValue = '';
+    if (
+      typeof this.value === 'boolean' ||
+      typeof this.value === 'number' ||
+      typeof this.value === 'bigint'
+    ) {
+      printableValue = String(this.value);
+    } else if (typeof this.value === 'string') {
+      printableValue = this.isInstance ? this.value : `"${fshifyString(this.value)}"`;
```

Comment:
It doesn't look like `fshifyString` handles the case that `this.value` has quotations in it. So the preprocessed string is written with un-escaped quotation marks.

Reply:
I think `fshifyString` handles this with [the second replacement](https://github.com/FHIR/sushi/pull/804/files#diff-6fca0c9ea8a1b6c0b3966479a503a1c9e7427f1074e9e2569f7862807fddcfb4R17). Also, if it weren't handling these strings correctly, wouldn't [this test for string values on an AssignmentRule](https://github.com/FHIR/sushi/pull/804/files#diff-1624e0922423d698734d0b7e59bdde701578748bce84498b3c1c44b3927d9a3aR37-R43) fail?

Reply:
Oh, I think I found what you're talking about: `CaretValueRule` is missing this usage of `fshifyString`. I'll add that in and update the test to prove it.